micro_ros_setup/README.md

```diff
@@ -1,80 +1,116 @@
+| [FreeRTOS](https://www.freertos.org/) | [Crazyflie 2.1](https://www.bitcraze.io/crazyflie-2-1/) | `freertos crazyflie21` | 
+| Linux / Windows | *Host** |
 
-We will use the standard ROS 2 meta build-tool, `colcon`, to compile. If you haven't used colcon, yet, you can probably just copy the instructions below as-is. I recommend [getting familiar with colcon](https://index.ros.org//doc/ros2/Tutorials/Colcon-Tutorial/), though, it has many useful options.
+*\* Support for compiling apps in native host for testing and debugging*
 
-## Concepts
+# Dependencies
 
-Micro-ROS is a client-server system: The embedded micro-controller runs the client, and the "agent" runs on Linux (or potentially also Windows, though this package doesn't support that, yet).
+This package targets **ROS 2 Dashing** installation. Some other prerequisites needed for building a firmware using this package are:
 
-Software on a microcontroller is usually called "firmware" and it is cross-compiled on the host and then flashed onto the microcontroller. The following structure image shows the individual parts:
-![structure image](doc/structure.png)
+```
+sudo apt install python-rosdep
+```
+
+# Building 
+
+Create a ROS 2 workspace and build this package:
 
-Correspondingly, in the micro-ROS build, we distinguish the firmware and the "regular" workspace for the server side. Since the regular workspace contains the agent, in other documentation it also sometimes called "agent workspace".
+```
+source /opt/ros/dashing/setup.bash
+mkdir uros_ws
+cd uros_ws
+git clone --recursive -b feature/multiboard https://github.com/micro-ROS/micro-ros-build.git src/micro-ros-build
+colcon build --packages-select micro_ros_setup
+source install/local_setup.bash
+```
 
-# Installation
+Once the package is built, the firmware scripts are ready to run.
 
-This is, overall, a four-step process.
 
-## 1/4 Setting up the workspace
+## Creating micro-ROS firmware
 
-It is best to start with a fresh workspace, both to speed up compiles and to avoid unintended interference. 
+Using `create_firmware_ws.sh [RTOS] [Platform]` command a firmware folder will be created with the required code for building a micro-ROS app. For example, targeting FreeRTOS and Crazyflie 2.1:
 
-```shell
-source /opt/ros/dashing/setup.bash
-sudo apt install python-rosdep
-mkdir -p uros_ws/src
-cd uros_ws
-git clone --recursive https://github.com/micro-ROS/micro-ros-build.git src/micro-ros-build
+```
+ros2 run micro_ros_setup create_firmware_ws.sh freertos crazyflie21
 ```
 
-All subsequent instructions assume that you're running them in the `uros_ws` directory. 
+## Configuring micro-ROS firmware
 
+By running `configure_firmware.sh` command the installed firmware is configured and modified in a pre-build step:
 
-## 2/4 Building this package
+```
+ros2 run micro_ros_setup configure_firmware.sh
```

Comment:
At least for NuttX builds, configure_firmware.sh would probaby take an argument specifying the configuration.

Reply:
Please check new changes